lib/src/index.rs

```diff
@@ -1955,6 +2004,91 @@ mod tests {
             PrefixResolution::AmbiguousMatch
         );
     }
+
+    #[test]
+    #[allow(clippy::redundant_clone)] // allow id_n.clone()
+    fn neighbor_commit_ids() {
+        let temp_dir = testutils::new_temp_dir();
+        let mut new_change_id = change_id_generator();
+        let mut index = MutableIndex::full(3);
+
+        // Create some commits with different various common prefixes.
+        let id_0 = CommitId::from_hex("000001");
+        let id_1 = CommitId::from_hex("009999");
+        let id_2 = CommitId::from_hex("055488");
+        index.add_commit_data(id_0.clone(), new_change_id(), &[]);
+        index.add_commit_data(id_1.clone(), new_change_id(), &[]);
+        index.add_commit_data(id_2.clone(), new_change_id(), &[]);
+
+        // Write the first three commits to one file and build the remainder on top.
+        let initial_file = index.save_in(temp_dir.path().to_owned()).unwrap();
+        index = MutableIndex::incremental(initial_file);
+
+        let id_3 = CommitId::from_hex("055444");
+        let id_4 = CommitId::from_hex("055555");
+        let id_5 = CommitId::from_hex("033333");
+        index.add_commit_data(id_3.clone(), new_change_id(), &[]);
+        index.add_commit_data(id_4.clone(), new_change_id(), &[]);
+        index.add_commit_data(id_5.clone(), new_change_id(), &[]);
+
+        // Local lookup in readonly index, commit_id exists.
+        let readonly_index = index.segment_parent_file().unwrap().as_ref();
```

Comment:
This is the same as `initial_file`, right? I think it's clearer to use that (and pass in a clone into `MutableIndex::incremental()` above).

Reply:
Indeed. Fixed, thanks.